bin/varnishd/cache/cache_vrt_vcl.c

```diff
@@ -274,6 +274,24 @@ VRT_LookupDirector(VRT_CTX, VCL_STRING name)
 
 /*--------------------------------------------------------------------*/
 
+VCL_BACKEND
+VRT_DirectorResolve(VRT_CTX, VCL_BACKEND d)
+{
+	VCL_BACKEND d2;
+
+	for (; d != NULL && d->vdir->methods->resolve != NULL; d = d2) {
+		CHECK_OBJ_NOTNULL(d, DIRECTOR_MAGIC);
+		AN(d->vdir);
+		d2 = d->vdir->methods->resolve(ctx, d);
+	}
```

Comment:
I'm a bit worried that we dereference `d->vdir->methods` before we check `d`. I'm also worried that we are painting ourselves in a corner if we expose this function with this signature. I will try to discuss this with you before adding more noise to this review.